rules/Php80/Rector/Class_/StringableForToStringRector.php

```diff
@@ -162,7 +162,7 @@ private function processNotStringType(ClassMethod $toStringClassMethod): void
             }
 
             $type = $this->nodeTypeResolver->getType($subNode->expr);
-            if ($type->isString()->yes()) {
+            if ($type->isVoid()->no()) {
```

Comment:
the solution should can check inner function instead, with verify early **before line 155** before `Return_` check:

```php
            if ($node instanceof \PhpParser\Node\Stmt\Class_ || $node instanceof \PhpParser\Node\Stmt\Function_ || $node instanceof \PhpParser\Node\Expr\Closure) {
                return \PhpParser\NodeTraverser::DONT_TRAVERSE_CURRENT_AND_CHILDREN;
            }
```

so it by pass inner function.